src/vs/workbench/contrib/terminal/browser/terminalTabsList.ts

```diff
@@ -621,6 +626,18 @@ class TerminalTabsDragAndDrop implements IListDragAndDrop<ITerminalInstance> {
 				if (instance) {
 					sourceInstances = [instance];
 					this._terminalService.moveToTerminalView(instance);
+				} else {
+					const processes = await this._localTerminalService.listProcesses(true);
+					this._localTerminalService.reduceConnectionGraceTime();
```

Comment:
Again a more specific call to detach would be better imo, we could avoid this `all` arg as well which simplifies things